config/dbstructure.php

```diff
@@ -0,0 +1,1374 @@
+<?php
+
+/* Main database structure configuration file.
+ *
+ * Here are described all the tables, fields and indexes Friendica needs to work.
+ *
+ * Syntax (braces indicate optionale values):
+ * "<table name>" => [
+ *	"comment" => "Description of the table",
+ *	"fields" => [
+ *		"<field name>" => [
+ *			"type" => "<field type>{(<field size>)} <unsigned>",
+ *			"not null" => 0|1,
+ *			{"extra" => "auto_increment",}
+ *			{"default" => "<default value>",}
+ *			{"default" => NULL_DATE,} (for datetime fields)
+ *			{"primary" => "1",}
+ *			{"relation" => ["<foreign key table name>" => "<foreign key field name>"],}
+ *			"comment" => "Description of the fields"
+ *		],
+ *		...
+ *	],
+ *	"indexes" => [
+ *		"PRIMARY" => ["<primary key field name>", ...],
+ *		"<index name>" => [{"UNIQUE",} "<field name>{(<key size>)}", ...]
+ *		...
+ *	],
+ * ],
+ *
+ * If you need to make any change, make sure to increment the DB_UPDATE_VERSION constant value below.
+ *
+ */
+
+use Friendica\Database\DBA;
+
+if (!defined('DB_UPDATE_VERSION')) {
```

Comment:
As the value is not defined anywhere else, why have you encapsulated the definition with the if block? Or is the value stored in the DB as well so it can come from there?

Reply:
In my late-night coding journey, this file could be required multiple times, which may have had defined the constant multiple times as well.

I later added a caching mechanism which should prevent multiple reads but I didn't remove the safeguard.